pkg/sql/sem/builtins/builtins.go

```diff
@@ -4927,6 +4928,32 @@ may increase either contention or retry errors, or both.`,
 		},
 	),
 
+	"crdb_internal.version_upgrade_type": makeBuiltin(
+		tree.FunctionProperties{Category: categorySystemInfo},
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"to_version", types.String},
+			},
+			ReturnType: tree.FixedReturnType(types.String),
+			Fn: func(evalCtx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				toVersion, err := version.Parse(string(tree.MustBeDString(args[0])))
```

Comment:
If we're looking to compare cluster versions, you can grab a handle on the current one through:

```go
			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
				cv := ctx.Settings.Version.ActiveVersion(ctx)
```